lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,380 @@
+
+    final nextButtonColor = selectedUserIds.isEmpty
+      ? designVariables.icon.withFadedAlpha(0.5)
+      : designVariables.icon;
+
+    return GestureDetector(
+      onTap: selectedUserIds.isEmpty ? null : () {
+        final narrow = DmNarrow.withUsers(
+          selectedUserIds.toList(),
+          selfUserId: store.selfUserId);
+        Navigator.pop(context);
+        Navigator.push(context,
+          MessageListPage.buildRoute(context: context, narrow: narrow));
+      },
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Text(zulipLocalizations.newDmSheetNextButtonLabel,
+            style: TextStyle(
+              color: nextButtonColor,
+              fontSize: 20,
+              height: 30 / 20,
+            ).merge(weightVariableTextStyle(context, wght: 600))),
+          Icon(ZulipIcons.chevron_right,
+            size: 24,
+            color: nextButtonColor),
+        ]));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return Container(
+      padding: const EdgeInsetsDirectional.fromSTEB(8, 10, 8, 6),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          _buildBackButton(context),
+          Expanded(
+            child: Text(zulipLocalizations.newDmSheetScreenTitle,
+              style: TextStyle(
+                color: designVariables.title,
+                fontSize: 20,
+                height: 30 / 20,
+              ).merge(weightVariableTextStyle(context, wght: 600)),
+              overflow: TextOverflow.ellipsis,
+              textAlign: TextAlign.center)),
+          _buildNextButton(context),
+        ]));
+  }
+}
+
+class _NewDmSearchBar extends StatelessWidget {
+  const _NewDmSearchBar({
+    required this.searchController,
+    required this.selectedUserIds,
+  });
+
+  final TextEditingController searchController;
+  final Set<int> selectedUserIds;
+
+  Widget _buildSearchField(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final hintText = selectedUserIds.isEmpty
+      ? zulipLocalizations.newDmSheetSearchHintEmpty
+      : zulipLocalizations.newDmSheetSearchHintSomeSelected;
+
+    return TextField(
+      controller: searchController,
+      cursorColor: designVariables.foreground,
+      style: TextStyle(
+        color: designVariables.textMessage,
+        fontSize: 17,
+        height: 22 / 17),
+      scrollPadding: EdgeInsets.zero,
+      decoration: InputDecoration(
+        isDense: true,
+        contentPadding: EdgeInsets.zero,
+        hintText: hintText,
+        hintStyle: TextStyle(
+          color: designVariables.labelSearchPrompt,
+          fontSize: 17,
+          height: 22 / 17),
+        border: InputBorder.none));
```

Comment:
nit: group properties logically; `border` is closely related to `contentPadding` (much more so than either of them is to the hint), so make them adjacent